maskrcnn_benchmark/structures/segmentation_mask.py

```diff
@@ -1,51 +1,97 @@
+            if type(segm) == list:
+                # polygons
+                rle = mask_utils.frPyObjects(segm, height, width)
+                mask = np.array(mask_utils.decode(rle), dtype=np.float32)
+                mask = np.sum(mask, axis=2)
+                mask = torch.from_numpy(np.array(mask > 0, dtype=np.float32))
+            elif type(segm) == dict and 'counts' in segm:
+                if type(segm['counts']) == list:
+                    # uncompressed RLE
+                    h, w = segm['size']
+                    rle = mask_utils.frPyObjects(segm, h, w)
+                    mask = mask_utils.decode(rle)
+                    mask = torch.from_numpy(mask).to(dtype=torch.float32)
+                else:
+                    # compressed RLE
+                    mask = mask_utils.decode(segm)
+                    mask = torch.from_numpy(mask).to(dtype=torch.float32)
+            else:
+                # binary mask
+                if type(segm) == np.ndarray:
+                    mask = torch.from_numpy(segm).to(dtype=torch.float32)
+                else: # torch.Tensor
+                    mask = segm
+        self.mask = mask
         self.size = size
         self.mode = mode
 
     def transpose(self, method):
         if method not in (FLIP_LEFT_RIGHT, FLIP_TOP_BOTTOM):
-            raise NotImplementedError(
-                "Only FLIP_LEFT_RIGHT and FLIP_TOP_BOTTOM implemented"
-            )
+            raise NotImplementedError("Only FLIP_LEFT_RIGHT and FLIP_TOP_BOTTOM implemented")
 
         width, height = self.size
         if method == FLIP_LEFT_RIGHT:
-            dim = width
-            idx = 2
+            max_idx = width
+            dim = 1
         elif method == FLIP_TOP_BOTTOM:
-            dim = height
-            idx = 1
+            max_idx = height
+            dim = 0
 
-        flip_idx = list(range(dim)[::-1])
-        flipped_masks = self.masks.index_select(dim, flip_idx)
-        return Mask(flipped_masks, self.size, self.mode)
+        flip_idx = torch.tensor(list(range(max_idx)[::-1]))
+        flipped_mask = self.mask.index_select(dim, flip_idx)
+        return Mask(flipped_mask, self.size, self.mode)
 
     def crop(self, box):
-        w, h = box[2] - box[0], box[3] - box[1]
-
-        cropped_masks = self.masks[:, box[1] : box[3], box[0] : box[2]]
-        return Mask(cropped_masks, size=(w, h), mode=self.mode)
+        box = [int(b) for b in box]
+        TO_REMOVE = 1
+        w, h = box[2] - box[0] + TO_REMOVE, box[3] - box[1] + TO_REMOVE
+        cropped_mask = self.mask[box[1]: box[3]+1, box[0]: box[2]+1]
+        return Mask(cropped_mask, size=(w, h), mode=self.mode)
 
+    # torch.nn.functional.interpolate has a arg as dim, only tensor have dim, so turn array to tensor
     def resize(self, size, *args, **kwargs):
-        pass
+        width, height = size
+        scaled_mask = torch.squeeze(interpolate(torch.from_numpy(np.array(self.mask)[None, None, :, :]).float(),
```

Comment:
Why do you perform this numpy conversion? torch tensors also support the `newaxis` style, and this can also be done via
```python
scaled_mask = interpolate(self.mask[None, None, :, :])[0, 0]
```